endpoint/domain_filter.go

```diff
@@ -146,35 +149,32 @@ func matchRegex(regex *regexp.Regexp, negativeRegex *regexp.Regexp, domain strin
 
 // MatchParent checks wether DomainFilter matches a given parent domain.
 func (df DomainFilter) MatchParent(domain string) bool {
-	if !df.IsConfigured() {
+	if matchFilter(df.exclude, domain, false) {
+		return false
+	}
+	if len(df.Filters) == 0 {
 		return true
 	}
 
+	strippedDomain := strings.ToLower(strings.TrimSuffix(domain, "."))
 	for _, filter := range df.Filters {
-		if strings.HasPrefix(filter, ".") {
+		if filter == "" || strings.HasPrefix(filter, ".") {
 			// We don't check parents if the filter is prefixed with "."
 			continue
 		}
-
-		if filter == "" {
-			return true
-		}
-
-		strippedDomain := strings.ToLower(strings.TrimSuffix(domain, "."))
-		if strings.HasSuffix(filter, "."+strippedDomain) && !matchFilter(df.exclude, domain, false) {
+		if strings.HasSuffix(filter, "."+strippedDomain) {
 			return true
 		}
 	}
-
 	return false
 }
 
-// IsConfigured returns true if DomainFilter is configured, false otherwise
+// IsConfigured returns true if any inclusion or exclusion rules have been specified.
 func (df DomainFilter) IsConfigured() bool {
```

Comment:
What is the purpose of modifying this method?

Reply:
There are call sites in the code that need to determine if filtering rules have been defined. As far as I can tell, those call sites are generally not concerned with "are allow rules defined" separately from "are exclude rules defined". As such, I've normalized this method to align with its name, "are any rules defined".